### PR TITLE
Allow columns to be sorted by highest material amount

### PR DESCRIPTION
**Pull request by LeonSkills**
Allows you to sort the columns by materials.  
Click on the "Qauntity" title to start sorting.  
Will keep the table up to date.  
Clicking on the "Comps" title will resume to regular behaviour.  

image

There is no visual indication that the sorting is on.. I am bad with css so couldn't figure out a nice way to do this.

### CODE REVIEW

File: scripts/script.js [hunk 1]
```diff
@@ -26,17 +37,14 @@ function readChatbox() {
     var chat = "";
     reader.find();
 
-    for (a in opts) {
+    for (a in opts.slice(-2)) {
```
Comment (LeonSkills) [inline scripts/script.js]: I don't think I was the one that changed this.. I'll change it back

**Comment by ZeroGwafa** [comment]
Could you please do another pull?  I had to fix a couple of issues just now.

**Comment by LeonSkills** [comment]
> 
> 
> Could you please do another pull? I had to fix a couple of issues just now.

Done!

File: scripts/data.js [hunk 2]
```diff
@@ -247,12 +247,12 @@ let materials = [
...
 if (localStorage.getItem("mats") != null) {
-    tempList = JSON.parse(localStorage.mats);
-    for (x in tempList) {
-        if (!tempList[x] || !materials[x]) {
-            continue;
-        } else {
-            materials[x].qty = tempList[x].qty;
-        }
-    }
+    materials = JSON.parse(localStorage.mats);
```
Comment (ZeroGwafa) [inline scripts/data.js:250]: I see what's going on here.  

This is what I get when:
A: Not just writing this from scratch, and just modifying my ComponentCounter Code to work for this new thing.
B: Doing this after being awake for 36 hours, and was supposed to sleep 8 hours prior.

**Review by ZeroGwafa — APPROVED** [review]
Gave it the once over.  Looks good! I have a couple more tweaks to be added once this is merged.

**Pull request merged**